python/lsst/daf/butler/script/queryDimensionRecords.py

```diff
@@ -19,31 +19,44 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+from __future__ import annotations
+
+from typing import Any
+
 from astropy.table import Table
 
 from .._butler import Butler
 from ..core import Timespan
 
 
-def queryDimensionRecords(repo, element, datasets, collections, where, no_check, order_by, limit, offset):
+def queryDimensionRecords(
+    repo: str,
+    element: str,
+    datasets: tuple[str, ...],
+    collections: tuple[str, ...],
+    where: str,
+    no_check: bool,
+    order_by: tuple[str, ...],
+    limit: int,
+    offset: int,
+) -> Table | None:
     # Docstring for supported parameters is the same as
     # Registry.queryDimensionRecords except for ``no_check``, which is the
     # inverse of ``check``.
 
     butler = Butler(repo)
 
-    records = butler.registry.queryDimensionRecords(
+    query_results = butler.registry.queryDimensionRecords(
         element, datasets=datasets, collections=collections, where=where, check=not no_check
     )
 
     if order_by:
-        records.order_by(*order_by)
+        query_results.order_by(*order_by)
```

Comment:
Hmm, not related to your changes, but I thought that we are supposed to do `query_results = query_results.order_by(*order_by)`, the fact that we modify iterator itself should be internal detail. Looking at documentation I think it's contradictory: https://github.com/lsst/daf_butler/blob/main/python/lsst/daf/butler/registry/queries/_results.py#L963-L972 
Same for `limit`, @TallJimbo, do you have an opinion on this?

Reply:
Definitely definitely return a new object.  DM-31725 is removing the modify-in-place, and making all of these objects immutable.

Reply:
Okay. Will fix.